File: miidas_aica_api/src/domain/public/master/spot_job_request.go
package master

// SpotJobRequestID スポット依頼内容ID
type SpotJobRequestID int

// SpotJobRequest スポット依頼内容
type SpotJobRequest struct {
	ID                    SpotJobRequestID    // ID
	Name                  string              // 依頼内容
	SpotJobRequestGenreID string              // ジャンルID
	Category              string              // カテゴリ
	SpotExpLevelPattern   SpotExpLevelPattern // 熟練度パターン
}

func (s SpotJobRequest) GetID() SpotJobRequestID {
	return s.ID
}

// TableName .
func (s SpotJobRequest) TableName() string {
	return "master.spot_job_request"
}

type (
	SpotJobRequests   = list[SpotJobRequestID, SpotJobRequest]
	SpotJobRequestMap = Map[SpotJobRequestID, SpotJobRequests]
)

File: miidas_aica_frontend/models/records/SpotJobRequest.ts
import { Record } from "immutable";

type Props = {
  /** スポット依頼内容ID */
  ID: number | null;
  /** 依頼内容 */
  Name: string;
  /** 熟練度パターン（"a" | "b" | "c" | "d" | "e"） */
  SpotExpLevelPattern: string;
};

/**
 * SpotJobRequest
 * スポット依頼内容レコード
 */
export default class SpotJobRequest extends Record<Props>({
  ID: null,
  Name: "",
  SpotExpLevelPattern: "",
}) {}
